cmd/otel-allocator/config/config.go

```diff
@@ -58,6 +59,79 @@ type Config struct {
 	HTTPS              HTTPSServerConfig     `yaml:"https,omitempty"`
 }
 
+func (cfg *Config) GetPrometheus() *monitoringv1.Prometheus {
+	// we want to use endpointslices by default
+	serviceDiscoveryRole := monitoringv1.ServiceDiscoveryRole("EndpointSlice")
+	prom := &monitoringv1.Prometheus{
+		Spec: monitoringv1.PrometheusSpec{
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				ScrapeInterval:                  monitoringv1.Duration(cfg.PrometheusCR.ScrapeInterval.String()),
+				ServiceMonitorSelector:          cfg.PrometheusCR.ServiceMonitorSelector,
+				PodMonitorSelector:              cfg.PrometheusCR.PodMonitorSelector,
+				ServiceMonitorNamespaceSelector: cfg.PrometheusCR.ServiceMonitorNamespaceSelector,
+				PodMonitorNamespaceSelector:     cfg.PrometheusCR.PodMonitorNamespaceSelector,
+				ServiceDiscoveryRole:            &serviceDiscoveryRole,
+			},
+		},
+	}
+
+	// if there's no prom config set, just use the above defaults.
+	if cfg.PromConfig == nil {
+		return prom
+	}
+
+	// the prometheus-operator provides no automatic conversion for global fields
+	if len(cfg.PromConfig.GlobalConfig.ScrapeProtocols) > 0 {
+		var scrapeProtocols []monitoringv1.ScrapeProtocol
+		for _, protocol := range cfg.PromConfig.GlobalConfig.ScrapeProtocols {
+			scrapeProtocols = append(scrapeProtocols, monitoringv1.ScrapeProtocol(protocol))
+		}
+		prom.Spec.CommonPrometheusFields.ScrapeProtocols = scrapeProtocols
```

Comment:
I don't think this is the correct thing to do. In my view, `GlobalConfig` should only affect the raw scrape configs, while the respective `Prometheus` fields (which only affect prometheus-operator CRs) should be separately configured.

The ambiguity of how `GlobalConfig` should affect the prometheus-operator world is why I was reluctant to include it in the first place.

Reply:
I can see that, however, i think the confusion / importance of doing it this way is that we don't have the luxury of having it automatically propagated to the prometheus instance. i.e. when a prometheus instance sets the global config and is being configured via the prometheus operator, any scrape configs generated by the prometheus CRDs to be added to the prometheus instance will be using the global config defined by said prometheus instance. 

As I was writing this, i was wondering if this is necessary given it's the collector that's doing the scraping. So if a user sets the global config on the prometheus receiver in the collector, shouldn't that be used when scraping a target overriding the scrape_configs received by the TA?

Reply:
I need to look into this, because if that's the case this would be unnecessary right? Otherwise, I _do_ think we need this.

Reply:
yeah that makes sense to me, and would probably better in the long run anyway. I had to look through some more code to make sense of this, but I agree that this is probably the way to go.